src/lavinmq/amqp/client.cr

```diff
@@ -410,23 +410,57 @@ module LavinMQ
         @vhost.rm_connection(self)
       end
 
-      private def close_socket
+      private def close_socket(force : Bool = false)
         @running = false
-        @socket.close
-        @log.debug { "Socket closed" }
+        if force && (socket = @socket.as?(Socket))
```

Comment:
for `OpenSSL::SSL::Socket` you can do `@socket.@bio.io.close_write`

Reply:
or monkey patch `OpenSSL::SSL::Socket`

Reply:
Oh, I actually forgot about that case!